resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/DriverCommandFeatureStep.scala

```diff
@@ -62,7 +62,18 @@ private[spark] class DriverCommandFeatureStep(conf: KubernetesDriverConf)
   }
 
   private def configureForJava(pod: SparkPod, res: String): SparkPod = {
-    val driverContainer = baseDriverContainer(pod, res).build()
+    val javaEnvs = if (conf.get(IS_PYTHON_APP)) {
+      Seq(new EnvVarBuilder()
+        .withName(ENV_PYSPARK_MAJOR_PYTHON_VERSION)
+        .withValue(conf.get(PYSPARK_MAJOR_PYTHON_VERSION))
+        .build())
+    } else {
+      Seq[EnvVar]()
+    }
+
+    val driverContainer = baseDriverContainer(pod, res)
+      .addAllToEnv(javaEnvs.asJava)
+      .build()
```

Comment:
Hi, @stczwd . This PR duplicates the code here and in `configureForPython`. Instead, can we refactor this duplicated logic into a common function?

Reply:
Yeap, glad to do that. Thanks for reply.